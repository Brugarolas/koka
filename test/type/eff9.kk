type foo :: X

external inline onevent : forall<e> ( () -> e () ) -> <foo|e> ()
{ inline "" }

function f() : <foo,foo> ()
{ function g() : foo ()
  {
    return ()
  }
  onevent(g)
}
