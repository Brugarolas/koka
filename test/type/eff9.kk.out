f : () -> <foo,foo> ()
onevent : forall<e> (() -> e ()) -> <foo|e> ()